ether-wake.c

```diff
@@ -109,17 +113,20 @@ static int get_nfqueue_num(const char *optarg);
 int main(int argc, char *argv[])
 {
 	char *ifname = "eth0";
+	char *ip_address;
 	int one = 1;				/* True, for socket options. */
 	int errflag = 0, nfqueue_errflag = 0, verbose = 0, do_version = 0;
 	int perm_failure = 0;
-	int i, c;
+	int i, c, ret;
 	struct ether_addr eaddr;
+	int(*send_function)() = &send_magic_packet;
 
 	while ((c = getopt(argc, argv, "bDi:p:q:uvV")) != -1)
```

Comment:
The getopt string needs to be updated to include the new option. Also, I didn't see any validation for the ip address argument, but never mind... Only a potential improvement.